virtual-repeater.js

```diff
@@ -321,9 +321,8 @@ export const Repeats = Superclass => class extends Superclass {
     const end = Math.min(this._last, this._prevFirst - 1);
     for (let idx = end; idx >= start; idx--) {
       const child = this._assignChild(idx);
-      if (!this._childIsAttached(child)) {
```

Comment:
I think this line is important for the searchable + invisible DOM case. Is there something different with lit-html that requires this?

Reply:
This guarantees the dom order is kept. It's important also for accessibility (e.g. tabbing through elements follows dom order) and consistent stacking context, as we're positioning all children with transform + position absolute, and we want the next siblings to render above their predecessors.

Reply:
example where not doing so breaks a11y & stacking context: user scrolls down, then scrolls back up. When scrolling back up:
1. we collect and recycle elements from the tail
2. we reuse these elements for the head

If we don't keep the dom order, the user would have weird order while tabbing through the items

Reply:
I think you're right about this being necessary for the truly virtualized (i.e. with recycling) case. I also think this breaks the searchable + invisible DOM case by always moving nodes that are being added to the visible range to either immediately before the last rendered range (here, using `this._insertBefore(child, this._firstChild);`) or the very end of the container (L342, using `this._insertBefore(child, null);`): it orders the rendered nodes correctly relative to each other but doesn't pay attention to anything that's not rendered and doesn't provide a way to prevent them from being moved, afaict. (The searchable + invisible DOM branches I had going relied on the virtual-scroller not moving nodes that were children of it.) Given that I haven't merged the searchable + invisible DOM branches, I still think this should go in, but I'm pretty sure we'll have to change it to fit that use.

Reply:
I see, then WDYT of restoring `_maintainDomOrder` property that was removed in #135? It could be configured in the constructor to false for the searchable + invisible DOM case

Reply:
Yeah, that might be a reasonable way to add that feature back. I wonder if that flag should be exposed to the user too. I'm going to go ahead and merge this given that it's more correct given the current state of the repo and shouldn't be much work to add back when needed.